awx_collection/plugins/modules/execution_environment.py

```diff
@@ -115,12 +115,14 @@ def main():
 
     # Attempt to look up the related items the user specified (these will fail the module if not found)
     organization = module.params.get('organization')
+    search_fields = {}
     if organization:
-        new_fields['organization'] = module.resolve_name_to_id('organizations', organization)
+        search_fields['organization'] = new_fields['organization'] = module.resolve_name_to_id('organizations', organization)
 
     credential = module.params.get('credential')
     if credential:
-        new_fields['credential'] = module.resolve_name_to_id('credentials', credential)
+        credential_object =  module.get_one('credentials', name_or_id=credential, **{'data': search_fields})
+        new_fields['credential'] = credential_object['id']
```

Comment:
This seems like it would be an anti-pattern to the lookup module we have going on. For example, f I want my EE in org1 to use a cred from org2. i.e. 
```
- execution_environmnet:
    organization: "My EE org"
    credential: "{{ lookup('awx.awx.tower_api', 'credential', { 'name': 'My Cred Name', 'organization': 'The org my cred belongs to'}) }}"
```

 I've already done the cred lookup with its org but now you are looking up the cred with an overridden org specified by the EE.
 
 Do you agree or am I thinking about this wrong?

Reply:
I think you are right, I think I may have just gone to far on this one. After writing it, and testing it, I was like, Maybe this is to far, enforcing this. I think it was more that was trying to tie things in, I am almost thinking a rewrite where you do things like
```
- execution_environmnet:
    organization: "My EE org"
    credential: 
        name: EE1
        organization: Org2
```
and lists for those that do that, so a different org can be supplied for every sub object.

Reply:
That could be its own rabbit hole as we would have to what pieces of information are required too lookup which objects.